### PR TITLE
Feature/cp 3432/sort by date

### PR DESCRIPTION
**Pull request by nf2000**
https://chillibean.atlassian.net/browse/CP-3432

### CODE REVIEW

File: nims-notes-things/src/components/Form.js [hunk 1]
```diff
@@ -58,53 +51,57 @@ const Form = (props) => {
...
+    emptyData.current[key].innerHTML =
+      formData[key] === "" ? `${key} required` : "";
   };
 
   return (
     <StyledForm onSubmit={handleOnSubmit}>
       <StyledDiv>
         <InnerForm>
-          {empty.name && <ErrorMessage>Name field is required</ErrorMessage>}
+          {notes.length === 0 && <p> Please Enter your Note</p>}
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: remove space after <p>

Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: The header of the **form** is only display when there are no notes, is that normal?

File: nims-notes-things/src/redux/actions/noteActions.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as noteActionTypes from "./noteActionTypes";
 
-export const setNote = (value) => {
+export const setNote = (value) => { // object formDataCopy from forms
```
Comment (damonspencerroberts) [inline nims-notes-things/src/redux/actions/noteActions.js]: move comment above line

File: nims-notes-things/src/components/notes.js [hunk 3]
```diff
@@ -1,37 +1,95 @@
...
-            <Note>
-              <p>{n.name}</p>
-              <p>{n.date}</p>
-              <p>{n.note}</p>
-            </Note>
-          </div>
-        );
-      })}
+      {notes.length !== 0 && (
+        <Button onClick={sortNote}>{sorted ? "unsort" : "sort"}</Button>
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/notes.js:51]: Very nice 👍

File: nims-notes-things/src/components/notes.js [hunk 4]
```diff
@@ -1,37 +1,95 @@
...
   });
 
+  const [sorted, setSorted] = useState(false);
+
+  const sortNote = (e) => {
+    e.preventDefault();
+    setSorted(!sorted);
+  };
+
+  const notemap = () => {
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/notes.js]: `notemap` => `noteMap` camel case it

File: nims-notes-things/src/components/notes.js [hunk 5]
```diff
@@ -1,37 +1,95 @@
...
 import { useSelector } from "react-redux";
+import Pin from "../pin.jpeg";
+import Button from "./Button";
 
 function Notes() {
   const notes = useSelector((state) => {
     return state.noteReducer.value;
   });
 
+  const [sorted, setSorted] = useState(false);
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/notes.js]: Told this Alex too, totally up to you, but try this with Redux for setting and handling the state. Would be a good challenge

**Review by damonspencerroberts — CHANGES_REQUESTED** [review]
Looking good, couple of small things!

File: nims-notes-things/src/components/Form.js (hunk 1 above)
Comment (nf2000) [reply nims-notes-things/src/components/Form.js]: Sorry this wasn't the header ... it was meant to be the message for when there are no notes submitted .... i've repositioned it and have also re-worded it

File: nims-notes-things/src/components/notes.js [hunk 6]
```diff
@@ -11,12 +11,17 @@ function Notes() {
 
   const [sorted, setSorted] = useState(false);
 
+  const dispatch = useDispatch();
+
+
   const sortNote = (e) => {
     e.preventDefault();
-    setSorted(!sorted);
+    //setSorted(!sorted);
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/notes.js]: remove

File: nims-notes-things/src/redux/reducers/noteReducer.js [hunk 7]
```diff
@@ -1,16 +1,20 @@
+import { setSorted } from "../actions/noteActions";
 import * as actions from "../actions/noteActionTypes";
 
 export const initialState = {
-  value: [], //whatever passed in - state arrary -> passed inside export function
+  //whatever passed in - state arrary -> passed inside export function
+  value: [],
```
Comment (damonspencerroberts) [inline nims-notes-things/src/redux/reducers/noteReducer.js:4]: seed to add sorted to this:
```
export const initialState = {
   sorted: false,
   value: []
}
```
`value` is and should only be used for the notes handling

File: nims-notes-things/src/redux/reducers/noteReducer.js [hunk 8]
```diff
@@ -1,16 +1,20 @@
...
 // eslint-disable-next-line import/no-anonymous-default-export
 export default function (state = initialState, action) {
   switch (action.type) {
     case actions.SET_NOTE:
-      const notesArray =  [...state.value]
+      const notesArray = [...state.value];
       notesArray.push(action.value);
       return { ...state, value: notesArray };
+    case actions.SET_SORTED:
+      return { ...state, value: setSorted };
```
Comment (damonspencerroberts) [inline nims-notes-things/src/redux/reducers/noteReducer.js]: this should be updating `sorted` field. When you update `value`, then you are updating the notes field

File: nims-notes-things/src/redux/reducers/noteReducer.js [hunk 9]
```diff
@@ -1,16 +1,20 @@
+import { setSorted } from "../actions/noteActions";
```
Comment (damonspencerroberts) [inline nims-notes-things/src/redux/reducers/noteReducer.js]: Why is this being imported?

File: nims-notes-things/src/components/notes.js [hunk 10]
```diff
@@ -11,12 +11,17 @@ function Notes() {
 
   const [sorted, setSorted] = useState(false);
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/notes.js]: No need for this state anymore if it is being handled with Redux

File: nims-notes-things/src/components/notes.js [hunk 11]
```diff
@@ -11,12 +11,17 @@ function Notes() {
...
-    setSorted(!sorted);
+    //setSorted(!sorted);
+    dispatch(setSorted(!sorted));
+
   };
 
-  const notemap = () => {
+  const noteMap = () => {
     let sortedData = [...notes];
     if (sorted) {
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/notes.js:26]: Where is sorted being imported from?

**Review by damonspencerroberts — CHANGES_REQUESTED** [review]
Need to change a couple things with the Redux logic. See Alex's example as we worked through it together, and he understands it pretty well. Very nearly there though 👍

File: nims-notes-things/src/components/notes.js (hunk 10 above)
Comment (nf2000) [reply nims-notes-things/src/components/notes.js]: Ahh - see now it makes sense why it was working lol i hadn't removed the line

File: nims-notes-things/src/components/Form.js [hunk 12]
```diff
@@ -7,6 +7,7 @@ import { v4 as uuidv4 } from "uuid";
 import { useSelector, useDispatch } from "react-redux";
 import { setNote } from "../redux/actions/noteActions";
 
+
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: remove whitespace

File: nims-notes-things/src/redux/reducers/noteReducer.js [hunk 13]
```diff
@@ -3,7 +3,7 @@ import * as actions from "../actions/noteActionTypes";
 
 export const initialState = {
   //whatever passed in - state arrary -> passed inside export function
-  value: [],
+  value: [], sorted : false
```
Comment (damonspencerroberts) [inline nims-notes-things/src/redux/reducers/noteReducer.js]: put on separate lines
```
{
   value: [],
   sorted: false
}

File: nims-notes-things/src/redux/reducers/noteReducer.js [hunk 14]
```diff
@@ -14,7 +14,8 @@ export default function (state = initialState, action) {
       notesArray.push(action.value);
       return { ...state, value: notesArray };
     case actions.SET_SORTED:
-      return { ...state, value: setSorted };
+      const set_sorted = state.sorted
```
Comment (damonspencerroberts) [inline nims-notes-things/src/redux/reducers/noteReducer.js]: semi colon

**Review by damonspencerroberts — APPROVED** [review]
Approved ☺️

**Pull request merged**